pg_check_indexes.rb

```diff
@@ -285,38 +285,29 @@ def foreign_key_check(connection_object, key_definition, do_stop)
   puts '*** START OF INDEX/TABLE CONSISTENCY TEST ***'
   puts
 
-  table_where =
-    if opts_sc[:tablename].nil?
-      ''
-    else
-      " AND pgn.nspname = '#{opts_sc[:tablename].split('.')[0]}' AND pgct.relname = \'"\
-        "#{opts_sc[:tablename].split('.')[1]}' "
-    end
-
   # The query will return a list of tables and their indexes to test, with additional info we need.
-  res_objects = conn_1.exec(
-    <<-SQL
-      SELECT DISTINCT
+  query = <<-SQL
+      SELECT
         pgn.nspname table_schema,
         pgct.relname table_name,
         pgci.relname index_name,
         pgi.indkey,
         pgi.indisunique,
         pga.amname index_type,
-        pg_relation_size (pgct.relname::regclass) table_size,
+        pg_relation_size (pgct.oid::regclass) table_size,
         pg_catalog.pg_get_indexdef(pgi.indexrelid, 0, true) index_definition
       FROM
-        (SELECT indkey, indisunique, indrelid,  indexrelid, unnest(indclass) indclasses FROM pg_index) pgi
+        pg_index pgi
         INNER JOIN pg_class pgct ON pgct.oid = pgi.indrelid
         INNER JOIN pg_class pgci ON pgci.oid = pgi.indexrelid
         INNER JOIN pg_namespace pgn ON pgn.oid=pgct.relnamespace
-        INNER JOIN pg_opclass pgo ON pgo.oid = pgi.indclasses::oid
-        INNER JOIN pg_am pga ON pga.oid = pgo.opcmethod::oid
+        INNER JOIN pg_am pga ON pga.oid = pgci.relam
```

Comment:
Using relam avoid needing to `unnest`ing the indclasses and then `DISTINCT`ing in the outer query.